packages/medusa/src/api/admin/claims/route.ts

```diff
@@ -0,0 +1,74 @@
+import { beginClaimOrderWorkflow } from "@medusajs/core-flows"
+import {
+  ContainerRegistrationKeys,
+  ModuleRegistrationName,
+  promiseAll,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../types/routing"
+import { AdminPostOrderClaimsReqSchemaType } from "./validators"
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse
+) => {
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "claims",
+    variables: {
+      filters: {
+        ...req.filterableFields,
+      },
+      ...req.remoteQueryConfig.pagination,
+    },
+    fields: req.remoteQueryConfig.fields,
+  })
+
+  const { rows: claims, metadata } = await remoteQuery(queryObject)
+
+  res.json({
+    claims,
+    count: metadata.count,
+    offset: metadata.skip,
+    limit: metadata.take,
+  })
+}
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<AdminPostOrderClaimsReqSchemaType>,
+  res: MedusaResponse
+) => {
+  const input = req.validatedBody as AdminPostOrderClaimsReqSchemaType
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const orderModuleService = req.scope.resolve(ModuleRegistrationName.ORDER)
+
+  const workflow = beginClaimOrderWorkflow(req.scope)
+  const { result } = await workflow.run({
+    input,
+  })
+
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "claim",
+    variables: {
+      id: result.claim_id,
+      filters: {
+        ...req.filterableFields,
+      },
+    },
+    fields: req.remoteQueryConfig.fields,
+  })
+
+  const [order, orderClaim] = await promiseAll([
+    orderModuleService.retrieveOrder(result.order_id),
```

Comment:
Yes, the Remote Joiner could resolve multiple queries in parallel if this is what you mean. Here is even simpler because it is a regular relation, but there are multiple flows where a single query could be used.
When we have all the flows done, I will take the time to cleanup all the repetition we have in most workflows where the validations are essencially the same thing and transform into single queries.